### DISCUSSION TITLE
Spacy stalls after its finished initializing

### DISCUSSION THREAD

**Discussion by aa-magic** [explosion/spaCy#9123] [Help: Coding & Implementations] [answered]
I'm trying to train a model and I first had to convert my data into the .spacy format
<img width="569" alt="Screen Shot 2021-09-02 at 8 16 10 PM" src="https://user-images.githubusercontent.com/89345997/132003244-45dcf3e4-8b73-4271-9675-6eb3dd5ac997.png">

I tried to combine all 50k files into 1 .spacy file, but that didn't work. Then I tried training by giving a directory with all the .spacy files instead of just 1. Yet after an hour of training, that came back as a format error for the training files. 

Is it possible to train on multiple different .spacy files instead of just one, and if so how do I do this?

**Comment by polm** [comment, marked as answer]
Please don't use screenshots of code/errors, just copy and paste text. 

By default all training data is loaded in memory, but it sounds like you have too much data for that. Look at the instructions for [streaming large or infinite corpora](https://spacy.io/usage/v3-1#streaming-corpora). 

However, if you have a format error that's something you'll have to deal with separately. What is your format error?

You might try using just a subset of your data to make sure everything is working, and then using the whole dataset.

**Comment by aa-magic** [comment]
I realized and fixed the issue - I did what you suggested regarding streaming large corpora. Thanks anyway!
